src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -1,10 +1,23 @@
 import {Logger, SfdxError} from '@salesforce/core';
-import {Catalog, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, ESRule, ESReport, ESMessage} from '../../types';
+import {Catalog, LooseObject, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, ESRule, ESReport, ESMessage} from '../../types';
 import {OutputProcessor} from '../pmd/OutputProcessor';
 import {RuleEngine} from '../services/RuleEngine';
 import {CLIEngine} from 'eslint';
 import * as path from 'path';
 
+// TODO: We recognize that this fix for W-7791882, including DEFAULT_ENV_VARS and the overrideDefaultEnv() method, is GARBAGE.
+//       During the 3.0 release cycle, we must replace it a better handling of the eslintrc file in general.
+// These are the environment variables that we'll want enabled by default in our ESLint baseConfig.
+let DEFAULT_ENV_VARS: LooseObject = {
+	'es6': true,           // `Map` class and others
+	'node': true,          // `process` global var and others
+	'browser': true,       // `document` global var
+	'webextensions': true  // Chrome
+};
+export function overrideDefaultEnv(overrideEnv: LooseObject): void {
+	DEFAULT_ENV_VARS = {...DEFAULT_ENV_VARS, ...overrideEnv};
```

Comment:
I like your innovative approach here. 

Also, wanted to point out that Mike had used an `Object.assign(sourceObject, destinationObject)` call to assign contents of one Json object into another. Using that approach, here we would do something like:
`Object.assign(DEFAULT_ENV_VARS, overrideEnv)` and DEFAULT_ENV_VARS would get all the new key/values.
However, for type definition, we were using Record<string, any> and had to add `@typescript-eslint/no-explicit-any` before each call.

Just wanted to lay this out here so that you had another path to consider and choose whichever you liked.